src/convenience_api.jl

```diff
@@ -0,0 +1,105 @@
+function GenericTrace{T<:Number,T2<:Number}(x::AbstractArray{T},
+                                            y::AbstractArray{T2};
+                                            kind="scatter",
+                                            kwargs...)
+    GenericTrace(kind; x=x, y=y, kwargs... )
+end
+
+"""
+$(SIGNATURES)
+Build a plot of with one trace of type `kind`and set `x` to x and `y` to y. All
+keyword arguments are passed directly as keyword arguments to the constructed
+trace.
+
+**NOTE**: If `y` is a matrix, one trace is constructed for each column of `y`
+
+**NOTE**: If `x` and `y` are both matrices, they must have the same number of
+columns (say `N`). Then `N` traces are constructed, where the `i`th column of
+`x` is paired with the `i`th column of `y`.
+"""
+function plot{T<:Number,T2<:Number}(x::AbstractArray{T}, y::AbstractArray{T2},
+                                    l::Layout=Layout(); kind="scatter", kwargs...)
+    plot(GenericTrace(x, y; kind=kind, kwargs...), l)
+end
+
+function plot{T<:Number,T2<:Number}(x::AbstractVector{T}, y::AbstractMatrix{T2},
+                                    l::Layout=Layout(); kwargs...)
+    traces = GenericTrace[GenericTrace(x, view(y, :, i); kwargs...)
+                          for i in 1:size(y,2)]
+    plot(traces, l)
+end
+
+function plot{T<:Number,T2<:Number}(x::AbstractMatrix{T}, y::AbstractMatrix{T2},
+                                    l::Layout=Layout(); kwargs...)
+    if size(x, 2) == 1
+        plot(view(x, :, 1), y, l; kwargs...)
+    end
+
+    size(x, 2) == size(y, 2) || error("x and y must have same number of cols")
+
+    traces = GenericTrace[GenericTrace(view(x, :, i), view(y, :, i); kwargs...)
+                          for i in 1:size(y,2)]
+    plot(traces, l)
+end
+
+"""
+$(SIGNATURES)
+Build a scatter plot and set  `y` to y. All keyword arguments are passed directly
+as keyword arguments to the constructed scatter.
+"""
+function plot{T<:Number}(y::AbstractArray{T}, l::Layout=Layout(); kwargs...)
+    # call methods above to get many traces if y is >1d
+    plot(1:size(y, 1), y, l; kwargs...)
+end
+
+"""
+$(SIGNATURES)
+Build a one `kind` trace plot using the layout `l` and all keyword arguments
+being passed to construct the trace.
+"""
+function plot(kind::Union{String,Symbol}, l::Layout=Layout(); kwargs...)
```

Comment:
As mentioned by @albep this function doesn't add much. You can do it using the other methods here by just passing the kind as a keyword argument. 

Towards the goal of keeping this API minimal and simple I'll delete it